modules/ckanext-ytp_main/ckanext/ytp/reports.py

```diff
@@ -99,16 +100,62 @@ def with_totals(orgs):
 }
 
 
-def package_generator(query, page_size, context):
-    package_search = get_action('package_search')
+def deprecated_datasets_report():
+    # Get packages that are deprecated
+    all_deprecated = package_generator('deprecated:true AND private:false', 1000, {})
+
+    # Function to loop packages through
+    # Get package visit and download data
+    # Resolve package structure to match table structure
+    def handle_package(pkg):
+        resolved_dict = {
+            'title': pkg["title"],
+            'id': pkg["id"],
+            'maintainer_name': pkg["maintainer"],
+            "maintainer_email": pkg["maintainer_email"],
+            'metadata_created': pkg["metadata_created"],
+            'valid_till': pkg["valid_till"],
+        }
 
-    for index in itertools.count(start=0, step=page_size):
-        data_dict = {'include_private': True, 'rows': page_size, 'q': query, 'start': index}
-        packages = package_search(context, data_dict).get('results', [])
-        for package in packages:
-            yield package
+        if pkg.get('organization'):
+            resolved_dict['organization_title'] = pkg["organization"].get("title")
+            resolved_dict['organization_homepage'] = pkg["organization"].get("homepage", None)
+            resolved_dict['organization_id'] = pkg["organization"].get("id")
+
+        # FIXME: Against CKAN best practices to access model directly, should be done through actions
```

Comment:
Was the reason for not using the best practice already now that we have used the similar bad practice often related to reports? So would we then change them all at the same time?

Reply:
Yeah that was what I was thinking. So there could be a PR to fix most of them at once.